src/level/level_loader.cpp

```diff
@@ -0,0 +1,1201 @@
+        { "TerrainCloud",       MakeCommand(&CLevelLoader::HandleCommand_TerrainCloud) },
+        { "TerrainBlitz",       MakeCommand(&CLevelLoader::HandleCommand_TerrainBlitz) },
+        { "TerrainInitTextures",MakeCommand(&CLevelLoader::HandleCommand_TerrainInitTextures) },
+        { "TerrainInit",        MakeCommand(&CLevelLoader::HandleCommand_TerrainInit) },
+        { "TerrainMaterial",    MakeCommand(&CLevelLoader::HandleCommand_TerrainMaterial) },
+        { "TerrainLevel",       MakeCommand(&CLevelLoader::HandleCommand_TerrainLevel) },
+        { "TerrainCreate",      MakeCommand(&CLevelLoader::HandleCommand_TerrainCreate) },
+        { "BeginObject",        MakeCommand(&CLevelLoader::HandleCommand_BeginObject, ALL_LOAD_LEVEL_MODES) },
+        { "LevelController",    MakeCommand(&CLevelLoader::HandleCommand_LevelController, LevelLoadMode::Normal | LevelLoadMode::Reset) },
+        { "CreateObject",       MakeCommand(&CLevelLoader::HandleCommand_CreateObject, LevelLoadMode::Normal | LevelLoadMode::Reset) },
+        { "CreateFog",          MakeCommand(&CLevelLoader::HandleCommand_CreateFog) },
+        { "CreateLight",        MakeCommand(&CLevelLoader::HandleCommand_CreateLight) },
+        { "CreateSpot",         MakeCommand(&CLevelLoader::HandleCommand_CreateSpot) },
+        { "GroundSpot",         MakeCommand(&CLevelLoader::HandleCommand_GroundSpot) },
+        { "WaterColor",         MakeCommand(&CLevelLoader::HandleCommand_WaterColor) },
+        { "MapColor",           MakeCommand(&CLevelLoader::HandleCommand_MapColor) },
+        { "MapZoom",            MakeCommand(&CLevelLoader::HandleCommand_MapZoom) },
+        { "MaxFlyingHeight",    MakeCommand(&CLevelLoader::HandleCommand_MaxFlyingHeight) },
+        { "AddFlyingHeight",    MakeCommand(&CLevelLoader::HandleCommand_AddFlyingHeight) },
+        { "Camera",             MakeCommand(&CLevelLoader::HandleCommand_Camera, ALL_LOAD_LEVEL_MODES) },
+        { "EndMissionTake",     MakeCommand(&CLevelLoader::HandleCommand_EndMissionTake) },
+        { "EndMissionDelay",    MakeCommand(&CLevelLoader::HandleCommand_EndMissionDelay) },
+        { "EndMissionResearch", MakeCommand(&CLevelLoader::HandleCommand_EndMissionResearch) },
+        { "ObligatoryToken",    MakeCommand(&CLevelLoader::HandleCommand_ObligatoryToken) },
+        { "ProhibitedToken",    MakeCommand(&CLevelLoader::HandleCommand_ProhibitedToken) },
+        { "EnableBuild",        MakeCommand(&CLevelLoader::HandleCommand_EnableBuild) },
+        { "EnableResearch",     MakeCommand(&CLevelLoader::HandleCommand_EnableResearch) },
+        { "DoneResearch",       MakeCommand(&CLevelLoader::HandleCommand_DoneResearch, LevelLoadMode::Normal) },
+        { "NewScript",          MakeCommand(&CLevelLoader::HandleCommand_NewScript) }
+};
+
+CLevelLoader::CLevelLoader(bool solutionEnabled)
+{
+    m_main = CRobotMain::GetInstancePointer();
```

Comment:
New code should really use dependency injection instead of the GetInstancePointer() functions whenever possible